src/Simplify.cpp

```diff
@@ -2146,6 +2129,16 @@ class Simplify : public IRMutator {
                    equal(min_a->b, min_b->b)) {
             // min(min(y, x), min(z, x)) -> min(min(y, z), x)
             expr = mutate(Min::make(Min::make(min_a->a, min_b->a), min_a->b));
+        } else if (max_a &&
+                   min_a_a &&
+                   equal(min_a_a->b, b)) {
+            // min(max(min(x, y), z), y) -> max(min(x, y), z)
```

Comment:
if z > y, this is wrong. Hrm... statically proving z > y is hard, and the constant case is already covered above. Maybe just cut this rule.